go/database/mpt/io/live.go

```diff
@@ -54,9 +54,16 @@ const (
 	EthereumHash = HashType(0)
 )
 
-// mptStateVisitor is an interface for Tries that allows for visiting the Trie nodes
+func NewExportableArchiveTrie(trie *mpt.ArchiveTrie, block uint64) MptStateVisitor {
+	return exportableArchiveTrie{
+		trie:  trie,
+		block: block,
+	}
+}
+
+// MptStateVisitor is an interface for Tries that allows for visiting the Trie nodes
 // and furthermore getting its properties such as a root hash and contract codes.
-type mptStateVisitor interface {
+type MptStateVisitor interface {
```

Comment:
There is not really a need to make this interface public.

If this interface is going to be public, I have higher quality expectations on its name.
The thing this interfaces defined is not a `Visitor`, so it shout not be named this way.

How about `VisitableState` or `ExportableState`?

Reply:
made private